### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by jmMeessen**
I propose with this pull request to add some unit tests to the codebase. With the help of Mockito, the dependencies are mocked.
While adding the tests for the various parameters, it appeared that some validations could be improved.

There is  still work to do to reach an acceptable unit test coverage but I wanted to submit this first batch to get some feedback.

### CODE REVIEW

**Comment by jmMeessen** [comment]
Hi @sanderv32 
Could you also have a look to this Pull Request ? I am making good progress on automated integration tests. Hope to be able to submit a first version in a week or so.

**Comment by sanderv32** [comment]
Certainly I will also have a look at this PR and merge it if the test doing their work. Many thanks for the effort! 🙂

**Comment by jmMeessen** [comment]
@sanderv32 I propose to rebase this PR tonight

**Comment by sanderv32** [comment]
@jmMeessen I already changed the GogsWebHook.java to the previous PR, so if you can have a look if everything is OK then we can merge this into master.

File: src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java [hunk 1]
```diff
@@ -88,128 +94,144 @@ public static String encode(String data, String key) throws Exception {
...
-            if (job != null) {
-                foundJob = true;
-				/* secret is stored in the properties of Job */
-				final GogsProjectProperty property = (GogsProjectProperty) job.getProperty(GogsProjectProperty.class);
-				if (property != null) { /* only if Gogs secret is defined on the job */
-					jSecret = property.getGogsSecret(); /* Secret provided by Jenkins */
-				}
+                if (job != null) {
+                    foundJob = true;
+                /* secret is stored in the properties of Job */
```
Comment (pascalw) [inline src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java]: Indentation of comments and code is out of place in several places.

Comment (jmMeessen) [inline src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java]: You're right @pascalw .
Note that this Pull Request will be closed and replaced by PR #26

**Pull request closed without merging**